jetty-security/src/main/java/org/eclipse/jetty/security/ConstraintSecurityHandler.java

```diff
@@ -434,12 +435,12 @@ protected void doStop() throws Exception
      */
     protected void processConstraintMapping(ConstraintMapping mapping)
     {
-        Map<String, RoleInfo> mappings = _constraintMap.get(mapping.getPathSpec());
+        Map<String, RoleInfo> mappings = _constraintMap.get(mapping.toPathSpec());
```

Comment:
This is a very subtle point here: mapping.toPathSpec() will return a PathSpec, but is being implicitly coerced to a String by the jvm .... if instead you called mapping.getPathSpec() you would get a string, but it would only be non-null if it was a ServletPathSpec. So I think this needs some good javadoc here, otherwise in 6months time someone could replace the toPathSpec() with getPathSpec()!

Reply:
Is this the case? _constraintMap is a PathMappings, so it takes a PathSpec?